db_whitebox_test.go

```diff
@@ -112,6 +119,181 @@ func TestMethodPage(t *testing.T) {
 	}
 }
 
+func TestTx_Check_CorruptPage(t *testing.T) {
+	testCases := []struct {
+		name      string
+		bucketKey string
+		isPanic   bool
+	}{
+		{
+			name:      "corrupt page by violating btree invariant",
+			bucketKey: "testBucket",
+			isPanic:   false,
+		},
+		{
+			name:      "corrupt page by dumping random bytes",
+			bucketKey: "testBucket",
+			isPanic:   true,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			fileName, err := prepareData(t)
+			require.NoError(t, err)
+
+			t.Logf("Creating db file '%v'.", fileName)
+			db, err := Open(fileName, 0666, &Options{PageSize: 4096})
+			require.NoError(t, err)
+			defer func() {
+				require.NoError(t, db.Close())
+			}()
+
+			uErr := db.Update(func(tx *Tx) error {
+				t.Logf("Creating bucket '%v'.", tc.bucketKey)
+				b, bErr := tx.CreateBucketIfNotExists([]byte(tc.bucketKey))
+				require.NoError(t, bErr)
+				t.Logf("Generating random data in bucket '%v'.", tc.bucketKey)
+				generateSampleDataInBucket(t, b, 3)
+				return nil
+			})
+			require.NoError(t, uErr)
+
+			t.Logf("Corrupting random leaf page in bucket '%v'.", tc.bucketKey)
+			victimPageId := corruptLeafPage(t, db, tc.isPanic)
+
+			if tc.isPanic {
```

Comment:
@ahrtr  so the panic I cause here is due to 
- writing random bytes between range `(16,4096]` 
- the panic is caused because the random bytes overwrites `leafPageElements.Pos` which is out of bound
- since the panic is not written to the `chan error`, there is no way i assert 

i dont know how to proceed 

cc @fuweid